tximeta in RNA-seq - Part 2: updating 03-gastric_cancer_exploratory.Rmd (#349)
AlexsLemonade/training-modules: RNA-seq/03-gastric_cancer_exploratory.Rmd

## DESeq2

> ### Creating a DESeq2 dataset from tximeta object

Reviewer: ```suggestion
### Creating a DESeq2 dataset from a tximeta object
```

First, let's read in the data we processed with `tximeta`.

```{r read-rds, live = TRUE}
# Read in the RDS file we created in the last notebook
gene_summarized <- readr::read_rds(txi_file)
```

### Set up DESeq2 object

DESeq2 wants data to be rounded. 

```{r}

> assay(gene_summarized) <- round(assay(gene_summarized))

Author: This feels a tad clumsy but 🤷‍♀️

Reviewer: This isn't necessary any more. DESeq2 handles unrounded data just fine.

Author: DESeq2 complained which is why I added this step. Will try again though.

We don't have batch information (i.e., when the samples were run) for this 
particular experiment, but let's imagine that `SRR585574` and `SRR585576` were
run separately from all other samples.

> We'll add this as a new "toy" column in the sample data (`colData`) for `gene_summarized`.

Reviewer: okay now that we are adding this to vst_data, let's make this slightly more general.
```suggestion
We'll add this as a new "toy" column in the sample data (`colData`).
```

We don't have batch information (i.e., when the samples were run) for this 
particular experiment, but let's imagine that `SRR585574` and `SRR585576` were
run separately from all other samples.
We'll add this as a new "toy" column in the sample data (`colData`) for `gene_summarized`.

```{r extract-sample, live = TRUE}
# Extract colData

> sample_info <- colData(gene_summarized)

Reviewer: For clarity with later, maybe we should pull the sample info from `vst_data`, as that is where we will be putting this back. They are identical, I think, but this might cause less confusion.
```suggestion
sample_info <- colData(vst_data)
```

Author: Ah. Yes. That would be less confusing.

# Print out preview
sample_info
```

Now we can add a new column with toy batch information and re-store the `colData()`. 

```{r add-batch}
# Add batch information
sample_info$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
                       "batch1", "batch2", "batch1")

# Replace the sample information                        

> colData(gene_summarized) <- sample_info

Reviewer: We aren't using this now, so we probably don't need to include this step: just adding it to `vst_data` (with a bit more explanation) should be fine.

run separately from all other samples.

> We'll put this in new "toy" column in `gene_summarized`.

Reviewer: ```suggestion
We'll add this as a new "toy" column in the sample data (`colData`) for `gene_summarized`.
```

We don't have batch information (i.e., when the samples were run) for this 
particular experiment, but let's imagine that `SRR585574` and `SRR585576` were
run separately from all other samples.
We'll put this in new "toy" column in `gene_summarized`.

```{r batches}

> gene_summarized@colData$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 

Author: Also feels a tad clumsy.

> gene_summarized@colData$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 

Author: Also feels a tad clumsy but I don't want to add this in the previous notebook since its fake data.

> gene_summarized@colData$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
>                           "batch1", "batch2", "batch1")

Reviewer: I think a better way to do this would be to do it in a few steps: start by pulling out the column data, then add the batch, then put it back in colData:
```suggestion
sample_info <- colData(gene_summarized)
sample_info$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
                          "batch1", "batch2", "batch1")
# replace the sample information                        
colData(gene_summarized) <- sample_info
```

One question I have is whether we need to actually do the next step of remaking the DESeq object. We could just put the colData into that object with the added batch column , and it would have the same effect (I think...). The only thing missing would be the comment about how you would "really" make the DESeq object if you had the batch information, but I am not sure that point comes across very well as it stands.

Author: Right. I think the idea is to show that you would need to remake the DESeq object because you *would* include batch in the equation *if* the batch info were real. But yeah, that message isn't super apparent.

Author: The reason we have to re-run this step is because of the `plotPCA()` step later that uses `batch` in the `intgroup` argument.

Author: Added some more explanation at this step. See how you feel about it.

Reviewer: > The reason we have to re-run this step is because of the `plotPCA()` step later that uses `batch` in the `intgroup` argument.

Right, I get that, but if we did `colData(vst_data) <- sample_info` it would work just fine, I think.

Author: Oh yeah, that seems like it would work. Will try that.

Author: That worked. And I *think* it's not more confusing now. See what you think.

> Now we need to rerun the `DESeq2` steps with this added toy batch column.

Reviewer: If we aren't going to rerun the steps, let's not say we will.
```suggestion
```

> If this batch information were real we would include it in the model stored in our DESeq2 object using the `design` argument. (`design = ~ tissue + batch`) and we would re-run the `DESeqDataSet()` and `vst()` steps we did above. 

Reviewer: ```suggestion
If this batch information were real we would have included it with the sample metadata when we made the original `SummarizedExperiment` object with `tximeta`.
We would then include it in the model stored in our DESeq2 object using the `design` argument (`design = ~ tissue + batch`) and we would re-run the `DESeqDataSet()` and `vst()` steps we did above. 
Here we will take a bit of a shortcut and add it directly to the `colData()` for our `vst()`-transformed data.
```
